mapdrawgdal.c

```diff
@@ -333,24 +333,32 @@ int msDrawRasterLayerGDAL(mapObj *map, layerObj *layer, imageObj *image,
 
   /*
    * Set up the band selection.  We look for a BANDS directive in the
-   * the PROCESSING options.  If not found we default to red=1 or
-   * red=1,green=2,blue=3 or red=1,green=2,blue=3,alpha=4.
+   * the PROCESSING options.  If not found we default to grey=1, grey=1,alpha=2,
+   * red=1,green=2,blue=3 or red=1,green=2,blue=3,alpha>=4.
    */
 
   if( CSLFetchNameValue( layer->processing, "BANDS" ) == NULL ) {
+    const int gdal_band_count = GDALGetRasterCount( hDS );
     red_band = 1;
 
-    if( GDALGetRasterCount( hDS ) >= 4
-        && GDALGetRasterColorInterpretation(
-          GDALGetRasterBand( hDS, 4 ) ) == GCI_AlphaBand )
-      alpha_band = 4;
+    if( gdal_band_count >= 4 )
+    {
+        /* The alpha band is not necessarily the 4th one */
+        for( i = 4; i <= gdal_band_count; i ++ ) {
+            if( GDALGetRasterColorInterpretation(
+                GDALGetRasterBand( hDS, i ) ) == GCI_AlphaBand ) {
```

Comment:
> From what I understood, (R,G,B,Unknown) was treated as (R,G,B,A) and would now be treated as (R,G,B)

I don't see anything in the code that would have done that and my below test also confirms that R,G,B,Unknown was treated as RGB.

Given test.map :
```
MAP

NAME TEST
STATUS ON
SIZE 400 300
EXTENT 0.5 0.5 399.5 299.5
IMAGECOLOR 255 255 0

IMAGETYPE png24_t

OUTPUTFORMAT
  NAME png24_t
  DRIVER "GDAL/PNG"
  IMAGEMODE RGBA
  TRANSPARENT ON
END

LAYER
  NAME rgba
  TYPE raster
  STATUS OFF
  DATA data/rgba.tif
END

LAYER
  NAME rgbu
  TYPE raster
  STATUS OFF
  DATA data/rgbunknown.tif
END


END # of map file
```

placed in msautotest/gdal. Where data/rgba.tif is the file in git and data/rgbunkown.tif is produced with the following operation :
```
gdal_translate data/rgba.tif data/rgbunknown.vrt -of vrt
edit data/rgbunknown.vrt  to remove <ColorInterp>Alpha</ColorInterp>
gdal_translate data/rgbunknown.vrt data/rgbunknown.tif  -co alpha=no
```

Then 'shp2img -m test.map -l rgba -o out_rgba.png' and 'shp2img -m test.map -l rgbu -o out_rgbu.png' produce different images. And that, before and after my patch.